src/FrozenSystemClock.cs

```diff
@@ -0,0 +1,142 @@
+﻿using System;
+
+namespace Archon
+{
+	/// <summary>
+	/// Frozen implementation of SystemClock
+	/// </summary>
+	public class FrozenSystemClock : SystemClock
+	{
+		DateTime frozen;
+
+		/// <summary>
+		/// Gets the current frozen time in UTC
+		/// </summary>
+		public DateTime UtcNow => frozen;
+
+		/// <summary>
+		/// Gets the current frozen time in local time
+		/// </summary>
+		public DateTime Now => frozen.ToLocalTime();
+
+		/// <summary>
+		/// Creates a new instance using the current UTC time as the frozen time
+		/// </summary>
+		public FrozenSystemClock()
+			: this(DateTime.UtcNow) { }
+
+		/// <summary>
+		/// Creates a new instance using the specified UTC frozen time
+		/// </summary>
+		/// <param name="frozenTimeUtc">The UTC time to freeze</param>
+		public FrozenSystemClock(DateTime frozenTimeUtc)
+		{
+			Reset(frozenTimeUtc);
+		}
+
+		/// <summary>
+		/// Reset the current frozen time to DateTime.UtcNow
+		/// </summary>
+		public void Reset()
+		{
+			Reset(DateTime.UtcNow);
+		}
+
+		/// <summary>
+		/// Reset the current frozen time to the specified UTC time
+		/// </summary>
+		/// <param name="frozenTimeUtc">The UTC time to freeze</param>
+		public void Reset(DateTime frozenTimeUtc)
+		{
+			if (frozenTimeUtc.Kind != DateTimeKind.Utc)
```

Comment:
why not just take the date and AssumeUtc() on it? saves consumer from having to remember to give a utc date